pkg/handler/triggerhandler.go

```diff
@@ -786,55 +813,62 @@ func request_PipelinePublicService_TriggerAsyncNamespacePipelineRelease_0_form(c
 }
 
 type streamingHandler struct {
-	writer http.ResponseWriter
-	ms     memory.MemoryStore
+	writer     http.ResponseWriter
+	subscriber pubsub.EventSubscriber
 }
 
-func NewStreamingHandler(writer http.ResponseWriter, ms memory.MemoryStore) *streamingHandler {
+func newStreamingHandler(writer http.ResponseWriter, sub pubsub.EventSubscriber) *streamingHandler {
 	return &streamingHandler{
-		writer: writer,
-		ms:     ms,
+		writer:     writer,
+		subscriber: sub,
 	}
 }
 
 func (sh *streamingHandler) Handle(ctx context.Context, triggerID string) {
-	wfm, err := sh.ms.GetWorkflowMemory(ctx, triggerID)
-	if err != nil {
-		return
-	}
-	defer func() {
-		_ = sh.ms.PurgeWorkflowMemory(ctx, triggerID)
```

Comment:
@donch1989 I removed this purge action here. Following the code, streaming endpoints call `TriggerAsync*` service usecases, so I added the same action [here](https://github.com/instill-ai/pipeline-backend/pull/981/files#diff-535874c672d3ba49bebb2add40574a260e061ba6df8286cd6c73eb3b8b51fe27R1851).

I noticed that there are many edge cases (mainly, non-happy paths) after the workflow memory has been created where it isn't cleared up. I didn't spend time fixing them, since we're moving away from the memory store.